pombola/core/tests/test_organisations.py

```diff
@@ -1,36 +1,143 @@
+
+
+def create_organisations(self):
+    self.test_organisation = Organisation.objects.create(
+        name="Test Org", slug="test-org", kind=self.foo_kind, ended="future"
+    )
+
+    self.na_organisation = Organisation.objects.create(
+        name="Basic Education", slug="basic-education", kind=self.na_kind,
+    )
+    self.ncop_organisation = Organisation.objects.create(
+        name="NCOP Appropriations",
+        slug="ncop-appropriations",
+        kind=self.ncop_kind,
+        ended="2010-01-01",
+    )
+
+
+def create_contact_kinds(self):
+    self.email_kind = ContactKind.objects.create(name="Email", slug="email")
+    self.phone_kind = ContactKind.objects.create(name="Phone", slug="phone")
+
+
+def create_contacts(self):
+    self.email_contact = Contact.objects.create(
+        kind=self.email_kind,
+        value="test@example.com",
+        content_type=ContentType.objects.get_for_model(self.na_organisation),
+        object_id=self.na_organisation.id,
+        preferred=True,
+    )
+    self.phone_contact = Contact.objects.create(
+        kind=self.phone_kind,
+        value="0000000",
+        content_type=ContentType.objects.get_for_model(self.na_organisation),
+        object_id=self.na_organisation.id,
+        preferred=True,
+    )
+
+
+@attr(country="south_africa")
+class OrganisationModelTest(TestCase):
     def setUp(self):
-        self.organisation_kind = models.OrganisationKind(
-            name = 'Foo',
-            slug = 'foo',
+        create_organisation_kinds(self)
+        create_organisations(self)
+        create_contact_kinds(self)
+        create_contacts(self)
+
+    def create_identifiers(self):
+        self.mysociety_id = Identifier.objects.create(
+            identifier="/organisations/1",
+            scheme="org.mysociety.za",
+            object_id=self.test_organisation.id,
+            content_type=ContentType.objects.get_for_model(Organisation),
         )
-        self.organisation_kind.save()
 
-        self.organisation = models.Organisation(
-            name = 'Test Org',
-            slug = 'test-org',
-            kind = self.organisation_kind,
+    def test_identifier(self):
+        self.create_identifiers()
+        org_mysociety_id = self.test_organisation.get_identifier("org.mysociety.za")
+        self.assertEqual(org_mysociety_id, "/organisations/1")
+
+    def test_email_addresses(self):
+        self.assertEqual(0, len(self.test_organisation.email_addresses))
+        na_contacts = self.na_organisation.email_addresses
+        self.assertEqual(1, len(na_contacts))
+        self.assertEqual([self.email_contact], na_contacts)
+        self.assertTrue(self.na_organisation.has_email_address)
+        self.assertFalse(self.test_organisation.has_email_address)
+
+    def test_is_committee(self):
+        self.assertTrue(self.na_organisation.is_committee)
+        self.assertTrue(self.ncop_organisation.is_committee)
+        self.assertFalse(self.test_organisation.is_committee)
```

Comment:
This comment is for this whole module, not just this test, but this test is a good example.

It's hard to understand the relevance between the data and the assertions. It's also brittle: if you wanna change data for a new test case, you have to look at all the other uses of that data. Someone might change something that doesn't make a test fail, but stops making the data relevant to the test.

The only way I know to improve this is to have data specific to the properties, (e.g. committee_organisation and non_committee_organisation) and being generous with the number of TestCase classes so that you don't set up 50 variables, and then list all the tests in one class. If committee_organisation _is_ relevant to multiple tests, you can see if they fit in the same class.

All the Organisation mode tests can be unit tests, using instances made wit the constructor without calling save(), I'm pretty sure

It should be possible to do the OrganisationQueryset tests as unit tests, but you might need a special implementation of Queryset to do that, so I think the test time incurred for just those tests should be tolerable?

Do you wanna do a quick example in the way I'm suggesting for me to check before doing everything?